src/chain/header.cpp

```diff
@@ -430,6 +430,8 @@ hash_digest header::hash() const
     mutex_.unlock_upgrade();
     ///////////////////////////////////////////////////////////////////////////
 
+    if (litecoin)
+        return litecoin_hash(to_data());
```

Comment:
For now, I've omitted here caching litecoin's pow hash because I wan't sure if that benefits performance.

Reply:
The benefits of caching the hash are significant. However it's not clear here whether the scrypt hash is used for anything other than PoW. If it is not then this method should not be modified. If it is then there will be a lot more impact.

Reply:
It's used for PoW exclusively.

Reply:
Then don't parameterize hash, just condition the PoW calc. Otherwise it will break the other stuff that hash is used for.